language/tools/test_generation/src/bytecode_generator.rs

```diff
@@ -44,47 +64,79 @@ enum BytecodeType {
-            BytecodeType::NoArg(Bytecode::GetTxnSequenceNumber),
+    pub fn new(seed: Option<[u8; 32]>) -> Self {
+        let instructions: Vec<(StackEffect, BytecodeType)> = vec![
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::Pop)),
+            (StackEffect::Add, BytecodeType::U64(Bytecode::LdConst)),
+            (
+                StackEffect::Add,
+                BytecodeType::StringPoolIndex(Bytecode::LdStr),
+            ),
+            (
+                StackEffect::Add,
+                BytecodeType::AddressPoolIndex(Bytecode::LdAddr),
+            ),
+            (StackEffect::Add, BytecodeType::NoArg(Bytecode::LdTrue)),
+            (StackEffect::Add, BytecodeType::NoArg(Bytecode::LdFalse)),
+            (
+                StackEffect::Add,
+                BytecodeType::ByteArrayPoolIndex(Bytecode::LdByteArray),
+            ),
+            (StackEffect::Add, BytecodeType::U8(Bytecode::CopyLoc)),
+            (StackEffect::Add, BytecodeType::U8(Bytecode::MoveLoc)),
+            (StackEffect::Sub, BytecodeType::U8(Bytecode::StLoc)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::Add)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::Sub)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::Mul)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::Div)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::Mod)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::BitAnd)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::BitOr)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::Xor)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::Or)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::And)),
+            (StackEffect::Nop, BytecodeType::NoArg(Bytecode::Not)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::Eq)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::Neq)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::Lt)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::Gt)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::Le)),
+            (StackEffect::Sub, BytecodeType::NoArg(Bytecode::Ge)),
+            (
+                StackEffect::Add,
+                BytecodeType::NoArg(Bytecode::GetTxnGasUnitPrice),
+            ),
+            (
+                StackEffect::Add,
+                BytecodeType::NoArg(Bytecode::GetTxnMaxGasUnits),
+            ),
+            (
+                StackEffect::Add,
+                BytecodeType::NoArg(Bytecode::GetGasRemaining),
+            ),
+            (
+                StackEffect::Add,
+                BytecodeType::NoArg(Bytecode::GetTxnSequenceNumber),
+            ),
         ];
-        let generator = StdRng::from_seed(seed);
+        let generator = match seed {
+            Some(seed) => StdRng::from_seed(seed),
+            None => {
```

Comment:
It seems bizarre to me that this non trivial bit of logic has not been packaged into a library routine.

Reply:
Adding a comment

Reply:
pub fn new() -> Result<StdRng>
Create a randomly seeded instance of StdRng.

This is a very expensive operation as it has to read randomness from the operating system and use this in an expensive seeding operation. If one is only generating a small number of random numbers, or doesn't need the utmost speed for generating each number, thread_rng and/or random may be more appropriate.

Reading the randomness from the OS may fail, and any error is propagated via the io::Result return value.

Reply:
The `new()` function you're describing is for a deprecated version of StdRng. While looking through the docs for our version, 0.6.5, I did come upon a trait called `FromEntropy` that allows us to initialize the StdRng without providing a seed.